templates/client-daemonset.yaml

```diff
@@ -235,6 +277,33 @@ spec:
           - name: aclconfig
             mountPath: /consul/aclconfig
       {{- end }}
+      {{- if .Values.global.tls.enabled }}
+      - name: client-tls-init
+        image: {{ .Values.global.image }}
+        env:
+        - name: HOST_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.hostIP
+        command:
+          - "/bin/sh"
+          - "-ec"
+          - |
+            cd /consul/tls/client
+            consul tls cert create -client \
+              -dc={{ .Values.global.datacenter }} \
+              -domain={{ .Values.global.domain }} \
+              -additional-ipaddress=${HOST_IP} \
```

Comment:
What is the use case for it? If we need it later for some reason, we could just add it to the cert, and it will get rotated on restart/upgrade. Eventually, I think we want to switch away from host IPs in favor of host-local traffic whenever kube supports it. In that case, we'd need to add a service name I think, but not sure what the actual implementation of it will be.

Reply:
One use-case would be some other type of tooling. Mostly it's just for completeness.

Reply:
I'm not sure this would be intended and have to think a bit more about security implications of this. I'll keep this out for now, but it should be easy to change if someone requests it.